templates/systems/components_selected.html

```diff
@@ -82,7 +79,11 @@ <h2 class="control-heading">
 <script>
   // Convert select field for Add Component to jQuery Select2 box
   $(document).ready(function() {
-      $('.producer_element_id').select2();
+    $('.producer_element_id').select2();
+  });
+  // Submit the add_component form when selection changed
+  $('.producer_element_id').on("select2:select", function(e) {
```

Comment:
This event is never being launched. If you add an "onchange" handler to the input form field in the form above, then write a function down here to call which just submits the form, that should do the trick. The field doesn't even need to be converted to a jQuery select2 field, according to my test.